src/components/blocks/Upvote.tsx

```diff
@@ -1,88 +1,86 @@
 import { AppwriteService } from "~/AppwriteService";
 import { AccountContext, UpvotesContext } from "~/routes/layout";
 
-export default component$(
-  (props: { projectId: string; votes: number; inline?: boolean }) => {
-    const location = useLocation();
-    const nav = useNavigate();
-    const upvoteContext = useContext(UpvotesContext);
-    const accountContext = useContext(AccountContext);
-    const isUpvoted = useComputed$(() => {
-      return upvoteContext.value.find(
-        (upvote) => upvote.projectId === props.projectId
-      )
-        ? true
-        : false;
-    });
+type Props = {
+  projectId: string;
+  votes: number;
+  inline?: boolean;
+};
 
-    const votes = useSignal(props.votes);
+export default component$((props: Props) => {
+  const upvoteContext = useContext(UpvotesContext);
+  const accountContext = useContext(AccountContext);
 
-    const isLoading = useSignal(false);
+  const isLoading = useSignal(false);
 
-    const upvoteProject = $(async () => {
-      isLoading.value = true;
-      try {
-        const res = await AppwriteService.upvoteProject(props.projectId);
+  const isUpvotedServer = useComputed$(() => {
+    return upvoteContext.value.find(
+      (upvote) => upvote.projectId === props.projectId
+    )
+      ? true
+      : false;
+  });
+  const isUpvotedClient = useSignal(isUpvotedServer.value);
+  const isUpvoted = useComputed$(() => {
+    return isLoading.value ? isUpvotedClient.value : isUpvotedServer.value;
+  });
 
-        if (accountContext.value) {
-          upvoteContext.value = await AppwriteService.listUserUpvotes(
-            accountContext.value.$id
-          );
-        }
-
-        votes.value = res.votes;
-        // isUpvoted.value = res.isUpvoted;
+  const votesServer = useSignal(props.votes);
+  const votes = useComputed$(() => {
+    if (isLoading.value) {
+      return isUpvotedClient.value
+        ? votesServer.value + 1
+        : votesServer.value - 1;
+    }
+    return votesServer.value;
+  });
 
-        const navUrl = location.url.pathname + location.url.search;
-        await nav(navUrl, true);
-      } catch (err: any) {
-        if (err.code && err.code === 401) {
-          alert("Please sign in first.");
-        } else {
-          alert(err.message);
-        }
-      } finally {
-        isLoading.value = false;
-      }
-    });
+  const upvoteProject = $(async (e: QwikMouseEvent) => {
+    e.stopPropagation();
 
-    let isPrimary = true;
+    isUpvotedClient.value = !isUpvotedServer.value;
+    isLoading.value = true;
 
-    if (isUpvoted.value === false || isLoading.value === true) {
-      isPrimary = false;
+    try {
+      const res = await AppwriteService.upvoteProject(props.projectId);
+      if (accountContext.value) {
+        upvoteContext.value = await AppwriteService.listUserUpvotes(
+          accountContext.value.$id
+        );
+      }
+      votesServer.value = res.votes;
+    } catch (err: any) {
```

Comment:
`any` is always kinda meh. I would start out with `unknown` 

```suggestion
    } catch (err: unknown) {
```

 If this comes from Appwrite, the SDK has an `AppwriteException` available as well 👍🏻